pkg-manager/resolve-dependencies/src/hoistPeers.ts

```diff
@@ -27,3 +27,20 @@ export function hoistPeers (
   }
   return dependencies
 }
+
+export function hoistOptionalPeers (
```

Comment:
nitpick: The name `hoistOptionalPeers` makes me think it'll perform a side-effect. In this case, it looks like it's just a pure function that computes a map. How about `getSatisfyingOptionalPeersFromFullGraph`?

Reply:
👏